packages/peregrine/lib/talons/WishlistPage/useWishlistItem.js

```diff
@@ -1,22 +1,40 @@
-import { useCallback } from 'react';
+import { useCallback, useState } from 'react';
 import { useMutation } from '@apollo/client';
 
-import { useCartContext } from '../../context/cart';
+import { useCartContext } from '@magento/peregrine/lib/context/cart';
+
+// Note: There is only ever zero (0) or one (1) dialogs open for a wishlist item.
+const dialogs = {
+    NONE: 1,
+    CONFIRM_REMOVE_PRODUCT: 2,
+    MORE_ACTIONS: 3
+};
 
 /**
  * @function
  *
  * @param {String} props.childSku SKU of the child item
+ * @param {String} props.itemId The ID of the item
  * @param {WishlistItemMutations} props.mutations GraphQL mutations for the Wishlist Item component
  * @param {String} props.sku SKU of the item
+ * @param {String} props.wishlistId The ID of the wishlist this item belongs to
  *
  * @returns {WishlistItemProps}
  */
 export const useWishlistItem = props => {
-    const { childSku, mutations, sku } = props;
-    const { addWishlistItemToCartMutation } = mutations;
+    const { childSku, itemId, mutations, sku, wishlistId } = props;
+
+    const {
+        addWishlistItemToCartMutation,
+        removeProductsFromWishlistMutation
+    } = mutations;
```

Comment:
Moved closer to the new colocation pattern in https://github.com/magento/pwa-studio/pull/2793/commits/a1dedb7ef5e80e73def409e073b2cdbdf06efc13 but since the queries relied on other fragments defined in `venia-ui` I left them there for now.